extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnPageReaderImpl.java

```diff
@@ -157,17 +162,16 @@ private int readRowCountFromDataPage(SeekableByteChannel file) throws IOExceptio
     }
 
     private IntBuffer readKeyFromDataPage(IntBuffer keyDest, int nullPlaceholder,
-            SeekableByteChannel file) throws IOException {
+            ReadableByteChannel file) throws IOException {
         int uncompressedPageSize = pageHeader.getUncompressed_page_size();
         int compressedPageSize = pageHeader.getCompressed_page_size();
 
         switch (pageHeader.type) {
             case DATA_PAGE:
-                ByteBuffer payload = Helpers.readFully(file, compressedPageSize);
-                DataPageHeader dataHeaderV1 = pageHeader.getData_page_header();
-                BytesInput decompressedInput = decompressorSupplier.get()
-                        .decompress(BytesInput.from(payload), pageHeader.getUncompressed_page_size());
+                BytesInput decompressedInput =
+                        compressor.decompress(Channels.newInputStream(file), compressedPageSize, uncompressedPageSize);
```

Comment:
Why are you skipping the buffering this time, vs. L142? Same question at L192, L219, and L237. I assume one pattern or the other is the "right" one.

Reply:
Whoops, buffering is now done inside the decompress method, and the buffering should be removed.